hack/e2e/deploy.yaml

```diff
@@ -0,0 +1,310 @@
+      - roleARN: <USER_ROLE>
+        username: "{{SessionNameRaw}}"
+        groups:
+        - pods-group
+      # each mapUsers entry maps an IAM role to a static username and set of groups
+      mapUsers:
+      - userARN: <USER_ARN>
+        username: currentuser
+        groups:
+        - nodes-group
+      # List of Account IDs to whitelist for authentication
+      mapAccounts:
+      - <AWS_ACCOUNT_ID>
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  namespace: kube-system
+  name: aws-iam-authenticator
+  labels:
+    k8s-app: aws-iam-authenticator
+  annotations:
+    seccomp.security.alpha.kubernetes.io/pod: runtime/default
+spec:
+  selector:
+    matchLabels:
+      k8s-app: aws-iam-authenticator
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      labels:
+        k8s-app: aws-iam-authenticator
+    spec:
+
+      # use service account with access to
+      serviceAccountName: aws-iam-authenticator
+
+      # run on the host network (don't depend on CNI)
+      hostNetwork: true
+
+      # run on each master node
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+      - key: CriticalAddonsOnly
+        operator: Exists
+
+      initContainers:
+      - name: aws-iam-authenticator-init
+        image: <IMAGE_NAME>
+        args:
+        - init
+        - -i <CLUSTER_NAME>
+        - --feature-gates=ConfiguredInitDirectories=true
+
+        securityContext:
+          privileged: true
+          runAsUser: 0
+          runAsGroup: 0
+
+        volumeMounts:
+        - name: config
+          mountPath: /etc/aws-iam-authenticator/
+        - name: state
+          mountPath: /var/aws-iam-authenticator/
+        - name: output
+          mountPath: /etc/kubernetes/aws-iam-authenticator/
+
+      - name: chown-helper
+        image: busybox
+        command: ["sh", "-c", "chown -R 10000:10000 /var/aws-iam-authenticator /etc/kubernetes/aws-iam-authenticator"]
+
+        securityContext:
+          privileged: true
+          runAsUser: 0
+          runAsGroup: 0
+
+        volumeMounts:
+        - name: state
+          mountPath: /var/aws-iam-authenticator/
+        - name: output
+          mountPath: /etc/kubernetes/aws-iam-authenticator/
+
+      # run `aws-iam-authenticator server` with three volumes
+      # - config (mounted from the ConfigMap at /etc/aws-iam-authenticator/config.yaml)
+      # - state (persisted TLS certificate and keys, mounted from the host)
+      # - output (output kubeconfig to plug into your apiserver configuration, mounted from the host)
+      containers:
+      - name: aws-iam-authenticator
+        image: <IMAGE_NAME>
+        args:
+        - server
+        # uncomment if using EKS-Style ConfigMap
+        # - --backend-mode=EKSConfigMap
```

Comment:
I think the default backend mode (the one without the flag, using a ConfigMap) is the one I'm using, so leaving this commented out is fine. I just left it in in case extra cases for different backend modes are added in the future.

Although now that I think about it, I should probably just get rid of it.